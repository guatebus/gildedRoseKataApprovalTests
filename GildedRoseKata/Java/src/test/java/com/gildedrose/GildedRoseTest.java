package com.gildedrose;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class GildedRoseTest {

  @Test
  void something() {
    String result = updateGildedRose("steelSword", 20, 5);

    //assertEquals(rose.items.length, items.length);
    Approvals.verify(result);
  }

  @Test
  void something_more() {
    String[] names = new String[]{
        "Aged Brie",
        "Backstage passes to a TAFKAL80ETC concert",
        "Sulfuras, Hand of Ragnaros",
        "applesauce"};
    Integer[] sellins = new Integer[]{4, 5, 10, 40, 0, -1};
    Integer[] qualities = new Integer[]{-1, 0, 10, 50, 51};

    CombinationApprovals.verifyAllCombinations(
        (name, sellin, quality) -> updateGildedRose(name, sellin, quality),
        names, sellins, qualities
    );
  }

  private String updateGildedRose(String name, int sellin, int quality) {
    Item[] x = new Item[] {new Item(name, sellin, quality)};

    GildedRose rose = new GildedRose(x);
    rose.updateQuality();

    return rose.items[0].toString();
  }
}