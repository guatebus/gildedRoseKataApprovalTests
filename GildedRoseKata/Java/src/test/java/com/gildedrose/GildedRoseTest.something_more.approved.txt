[Aged Brie, 4, -1] => Aged Brie, 3, 0 
[Aged Brie, 4, 0] => Aged Brie, 3, 1 
[Aged Brie, 4, 10] => Aged Brie, 3, 11 
[Aged Brie, 4, 50] => Aged Brie, 3, 50 
[Aged Brie, 4, 51] => Aged Brie, 3, 51 
[Aged Brie, 5, -1] => Aged Brie, 4, 0 
[Aged Brie, 5, 0] => Aged Brie, 4, 1 
[Aged Brie, 5, 10] => Aged Brie, 4, 11 
[Aged Brie, 5, 50] => Aged Brie, 4, 50 
[Aged Brie, 5, 51] => Aged Brie, 4, 51 
[Aged Brie, 10, -1] => Aged Brie, 9, 0 
[Aged Brie, 10, 0] => Aged Brie, 9, 1 
[Aged Brie, 10, 10] => Aged Brie, 9, 11 
[Aged Brie, 10, 50] => Aged Brie, 9, 50 
[Aged Brie, 10, 51] => Aged Brie, 9, 51 
[Aged Brie, 40, -1] => Aged Brie, 39, 0 
[Aged Brie, 40, 0] => Aged Brie, 39, 1 
[Aged Brie, 40, 10] => Aged Brie, 39, 11 
[Aged Brie, 40, 50] => Aged Brie, 39, 50 
[Aged Brie, 40, 51] => Aged Brie, 39, 51 
[Aged Brie, 0, -1] => Aged Brie, -1, 1 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 10] => Aged Brie, -1, 12 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 0, 51] => Aged Brie, -1, 51 
[Aged Brie, -1, -1] => Aged Brie, -2, 1 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 10] => Aged Brie, -2, 12 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, -1, 51] => Aged Brie, -2, 51 
[Backstage passes to a TAFKAL80ETC concert, 4, -1] => Backstage passes to a TAFKAL80ETC concert, 3, 2 
[Backstage passes to a TAFKAL80ETC concert, 4, 0] => Backstage passes to a TAFKAL80ETC concert, 3, 3 
[Backstage passes to a TAFKAL80ETC concert, 4, 10] => Backstage passes to a TAFKAL80ETC concert, 3, 13 
[Backstage passes to a TAFKAL80ETC concert, 4, 50] => Backstage passes to a TAFKAL80ETC concert, 3, 50 
[Backstage passes to a TAFKAL80ETC concert, 4, 51] => Backstage passes to a TAFKAL80ETC concert, 3, 51 
[Backstage passes to a TAFKAL80ETC concert, 5, -1] => Backstage passes to a TAFKAL80ETC concert, 4, 2 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Backstage passes to a TAFKAL80ETC concert, 4, 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 10] => Backstage passes to a TAFKAL80ETC concert, 4, 13 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 51] => Backstage passes to a TAFKAL80ETC concert, 4, 51 
[Backstage passes to a TAFKAL80ETC concert, 10, -1] => Backstage passes to a TAFKAL80ETC concert, 9, 1 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert, 9, 2 
[Backstage passes to a TAFKAL80ETC concert, 10, 10] => Backstage passes to a TAFKAL80ETC concert, 9, 12 
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => Backstage passes to a TAFKAL80ETC concert, 9, 50 
[Backstage passes to a TAFKAL80ETC concert, 10, 51] => Backstage passes to a TAFKAL80ETC concert, 9, 51 
[Backstage passes to a TAFKAL80ETC concert, 40, -1] => Backstage passes to a TAFKAL80ETC concert, 39, 0 
[Backstage passes to a TAFKAL80ETC concert, 40, 0] => Backstage passes to a TAFKAL80ETC concert, 39, 1 
[Backstage passes to a TAFKAL80ETC concert, 40, 10] => Backstage passes to a TAFKAL80ETC concert, 39, 11 
[Backstage passes to a TAFKAL80ETC concert, 40, 50] => Backstage passes to a TAFKAL80ETC concert, 39, 50 
[Backstage passes to a TAFKAL80ETC concert, 40, 51] => Backstage passes to a TAFKAL80ETC concert, 39, 51 
[Backstage passes to a TAFKAL80ETC concert, 0, -1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 10] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 51] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, -1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 10] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 51] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Sulfuras, Hand of Ragnaros, 4, -1] => Sulfuras, Hand of Ragnaros, 4, -1 
[Sulfuras, Hand of Ragnaros, 4, 0] => Sulfuras, Hand of Ragnaros, 4, 0 
[Sulfuras, Hand of Ragnaros, 4, 10] => Sulfuras, Hand of Ragnaros, 4, 10 
[Sulfuras, Hand of Ragnaros, 4, 50] => Sulfuras, Hand of Ragnaros, 4, 50 
[Sulfuras, Hand of Ragnaros, 4, 51] => Sulfuras, Hand of Ragnaros, 4, 51 
[Sulfuras, Hand of Ragnaros, 5, -1] => Sulfuras, Hand of Ragnaros, 5, -1 
[Sulfuras, Hand of Ragnaros, 5, 0] => Sulfuras, Hand of Ragnaros, 5, 0 
[Sulfuras, Hand of Ragnaros, 5, 10] => Sulfuras, Hand of Ragnaros, 5, 10 
[Sulfuras, Hand of Ragnaros, 5, 50] => Sulfuras, Hand of Ragnaros, 5, 50 
[Sulfuras, Hand of Ragnaros, 5, 51] => Sulfuras, Hand of Ragnaros, 5, 51 
[Sulfuras, Hand of Ragnaros, 10, -1] => Sulfuras, Hand of Ragnaros, 10, -1 
[Sulfuras, Hand of Ragnaros, 10, 0] => Sulfuras, Hand of Ragnaros, 10, 0 
[Sulfuras, Hand of Ragnaros, 10, 10] => Sulfuras, Hand of Ragnaros, 10, 10 
[Sulfuras, Hand of Ragnaros, 10, 50] => Sulfuras, Hand of Ragnaros, 10, 50 
[Sulfuras, Hand of Ragnaros, 10, 51] => Sulfuras, Hand of Ragnaros, 10, 51 
[Sulfuras, Hand of Ragnaros, 40, -1] => Sulfuras, Hand of Ragnaros, 40, -1 
[Sulfuras, Hand of Ragnaros, 40, 0] => Sulfuras, Hand of Ragnaros, 40, 0 
[Sulfuras, Hand of Ragnaros, 40, 10] => Sulfuras, Hand of Ragnaros, 40, 10 
[Sulfuras, Hand of Ragnaros, 40, 50] => Sulfuras, Hand of Ragnaros, 40, 50 
[Sulfuras, Hand of Ragnaros, 40, 51] => Sulfuras, Hand of Ragnaros, 40, 51 
[Sulfuras, Hand of Ragnaros, 0, -1] => Sulfuras, Hand of Ragnaros, 0, -1 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 10] => Sulfuras, Hand of Ragnaros, 0, 10 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, 0, 51] => Sulfuras, Hand of Ragnaros, 0, 51 
[Sulfuras, Hand of Ragnaros, -1, -1] => Sulfuras, Hand of Ragnaros, -1, -1 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 10] => Sulfuras, Hand of Ragnaros, -1, 10 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[Sulfuras, Hand of Ragnaros, -1, 51] => Sulfuras, Hand of Ragnaros, -1, 51 
[applesauce, 4, -1] => applesauce, 3, -1 
[applesauce, 4, 0] => applesauce, 3, 0 
[applesauce, 4, 10] => applesauce, 3, 9 
[applesauce, 4, 50] => applesauce, 3, 49 
[applesauce, 4, 51] => applesauce, 3, 50 
[applesauce, 5, -1] => applesauce, 4, -1 
[applesauce, 5, 0] => applesauce, 4, 0 
[applesauce, 5, 10] => applesauce, 4, 9 
[applesauce, 5, 50] => applesauce, 4, 49 
[applesauce, 5, 51] => applesauce, 4, 50 
[applesauce, 10, -1] => applesauce, 9, -1 
[applesauce, 10, 0] => applesauce, 9, 0 
[applesauce, 10, 10] => applesauce, 9, 9 
[applesauce, 10, 50] => applesauce, 9, 49 
[applesauce, 10, 51] => applesauce, 9, 50 
[applesauce, 40, -1] => applesauce, 39, -1 
[applesauce, 40, 0] => applesauce, 39, 0 
[applesauce, 40, 10] => applesauce, 39, 9 
[applesauce, 40, 50] => applesauce, 39, 49 
[applesauce, 40, 51] => applesauce, 39, 50 
[applesauce, 0, -1] => applesauce, -1, -1 
[applesauce, 0, 0] => applesauce, -1, 0 
[applesauce, 0, 10] => applesauce, -1, 8 
[applesauce, 0, 50] => applesauce, -1, 48 
[applesauce, 0, 51] => applesauce, -1, 49 
[applesauce, -1, -1] => applesauce, -2, -1 
[applesauce, -1, 0] => applesauce, -2, 0 
[applesauce, -1, 10] => applesauce, -2, 8 
[applesauce, -1, 50] => applesauce, -2, 48 
[applesauce, -1, 51] => applesauce, -2, 49 
